src/ToolUpdate/UpdateExecutor.php

```diff
@@ -37,17 +40,30 @@ final class UpdateExecutor
      */
     private $platform;
 
+    /**
+     * @var SignatureVerifier
+     */
+    private $verifier;
+
+    /**
+     * @var callable
+     */
+    private $untrustedKeyStrategy;
+
     public function __construct(
         PlatformInformationInterface $platform,
         FileDownloader $downloader,
+        SignatureVerifier $verifier,
         string $phpcqPath,
-        OutputInterface $output
-    )
-    {
-        $this->platform   = $platform;
-        $this->downloader = $downloader;
-        $this->phpcqPath  = $phpcqPath;
-        $this->output     = $output;
+        OutputInterface $output,
+        callable $untrustedKeyStrategy
```

Comment:
Can't we shift the strategy into the verifier? That way we would not couple this class with yet another one.

IMO fetching and accepting keys on the fly makes more sense in the verifier than in the updater even from a logical point of view. WDYT?

Reply:
IMO I think that the pgp related stuff could be extracted into an own library. But even if we do it, we can use a wrapper verifier handling the strategy.

Reply:
Done with 1dd7239.